services/QuillLMS/engines/comprehension/app/models/comprehension/feedback.rb

```diff
@@ -20,5 +22,31 @@ def serializable_hash(options = nil)
         include: [:highlights]
       ))
     end
+
+    private def semantic_rule
```

Comment:
Not that important, but it's Rails convention to name functions that provide boolean responses with `?` at the end.  So this, and the next two functions, should probably be `def semantic_rule?`, `def first_order?`, and `def second_order?`.